File: fe/ClinicManager/src/models/audit_log.ts
export enum AuditAction {
    CREATE = "CREATE",
    UPDATE = "UPDATE",
    DELETE = "DELETE"
}

export interface AuditLog {
    id: number;
    entityType: string;
    entityId: number;
    entityKeyTrait: string;
    action : AuditAction;
    performedBy: string;
    timestamp: string;
    oldValue: string | null;
    newValue: string | null;
    changedFields: string;
    ipAddress: string;
    sessionId: string;
    deviceType: string;
    browserName: string;
}

export interface AuditLogFilterDTO {
    entityType: string | null;
    keyword: string | null;
    action : AuditAction | null;
    performedBy: string | null;
    dateFrom: string | null;
    dateTo: string | null;
}

File: be/ClinicManager/src/main/java/com/clinic/clinicmanager/model/AuditLog.java
package com.clinic.clinicmanager.model;

import com.clinic.clinicmanager.model.constants.AuditAction;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "audit_log")
public class AuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String entityType;

    @Column(nullable = false)
    private Long entityId;

    @Column(nullable=true)
    private String entityKeyTrait;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AuditAction action;

    @Column(nullable = false)
    private String performedBy;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @Column(columnDefinition = "TEXT")
    private String oldValue;

    @Column(columnDefinition = "TEXT")
    private String newValue;

    @Column
    private String changedFields;

    @Column
    private String ipAddress;

    @Column
    private String sessionId;

    @Column
    private String deviceType;

    @Column
    private String browserName;
}
